### PR TITLE
Added testing instructions

### LINKED ISSUE #2959: Update GETTING-STARTED.md
I created a new tag. ingame-testing-needed. The name is self explanatory. ~~It is used for PRs only~~. It should be explained in "getting started" as it is great for beginners, just like the "easy" tag.

Edit: The idea is that maintainers put it on PRs, and essentially it's a "please test this, people" request. non-members can pull it, test it, and report back.

### PR DESCRIPTION
**Pull request by LogicParrot**
Closes #2959

### CODE REVIEW

**Comment by LogicParrot** [comment]
Do the CIs retain the binaries and allow them to be downloadable? And for how long? If not, I may need to modify the instructions a bit.

**Comment by bearbin** [comment]
Travis does not retain binaries. I don't believe appveyor retains them either.

builds.cuberite.org retains binaries indefinitely.

File: TESTING.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+----------------------
+
+When testing Cuberite, debug builds are preferred because they output detailed info about the problem that had occured at the server console.
+Currently, we do not provide debug builds. You will have to compile Cuberite yourself to obtain them. See the compilation instructions.
+You can also test with regular builds.
+
+Testing unmerged pull requests
+------------------------------
+
+To test unmerged branches, pick one from the [ingame-testing-needed](https://github.com/cuberite/cuberite/pulls?q=is%3Aopen+is%3Apr+label%3Aingame-testing-needed) list, and follow the download instructions.
```
Comment (LogicParrot) [inline TESTING.md]: Todo change this line.

**Comment by LogicParrot** [comment]
Can you easily make builds.cuberite.org compile selected PRs?

**Comment by bearbin** [comment]
I could make it compile all PRs if necessary. Individual ones might be harder. 

I could also build all commits on a particular branch, updating that branch to point to the PR might be an easier way to only build certain ones.

**Comment by LogicParrot** [comment]
> I could also build all commits on a particular branch, updating that branch to point to the PR might be an easier way to only build certain ones.

Sounds optimal to me. I can take care of maintaining it by pushing/reverting stuff appropriately.

**Comment by bearbin** [comment]
@LogicParrot Great. Any changes pushed to branch `pr-testing` will be built to Linux x64 and Windows x64.

https://builds.cuberite.org/job/Cuberite%20Linux%20x64%20pr-testing/
https://builds.cuberite.org/job/Cuberite%20Windows%20x64%20pr-testing/

**Comment by LogicParrot** [comment]
If I revert and then push totally unrelated changes, the buildserver wouldn't care, right? It'd just rebuild the new stuff whilst retaining a download link for the old stuff. Am I correct?

**Comment by bearbin** [comment]
That's right.

**Comment by LogicParrot** [comment]
Great. I've pushed something already, seems to be working :)

File: TESTING.md (hunk 1 above)
Comment (LogicParrot) [inline TESTING.md]: No need to change the line.

**Comment by LogicParrot** [comment]
Is it available for anything other than LinuxX64 and win64?
I think it should be enabled for all platforms. Since the compiles are going to be very infrequent, this shouldn't be a problem performance-wise.

**Comment by LogicParrot** [comment]
The name `pr-testing` may be confused with `testing`.

**Comment by LogicParrot** [comment]
Oh, one more thing: These builds should prefereablly be debug mode.

**Comment by bearbin** [comment]
I can put them in debug mode. What would be your preferred name for the branch?

I don't know about all platforms, it shouldn't be a performance problem, just maintenance would take longer. I'll do it on raspi too, would that be good?

**Comment by LogicParrot** [comment]
@bearbin Ok. I guess Rpi + Lin/win 64 cover most of the audience.
`ingame-testing-needed` sounds good and consistent with the tag.

**Comment by bearbin** [comment]
Yes that sounds good. I'll change the name/branch that it builds from.

**Comment by LogicParrot** [comment]
@bearbin We have a strange failure. Does it have to do with setting up debug builds? (and trying to execute ./Cuberite rather than ./Cuberite_debug?)

https://builds.cuberite.org/job/Cuberite%20Linux%20x64%20ingame-testing-needed/2/console

**Comment by bearbin** [comment]
That was the problem. I've fixed it now.

**Pull request merged**